src/pages/home/report/withReportAndPrivateNotesOrNotFound.js

```diff
@@ -0,0 +1,129 @@
+import React, {useEffect, useMemo} from 'react';
+import PropTypes from 'prop-types';
+import _ from 'underscore';
+import lodashGet from 'lodash/get';
+import {withOnyx} from 'react-native-onyx';
+import getComponentDisplayName from '../../../libs/getComponentDisplayName';
+import reportPropTypes from '../../reportPropTypes';
+import withReportOrNotFound from './withReportOrNotFound';
+import * as Report from '../../../libs/actions/Report';
+import NotFoundPage from '../../ErrorPage/NotFoundPage';
+import FullScreenLoadingIndicator from '../../../components/FullscreenLoadingIndicator';
+import {withNetwork} from '../../../components/OnyxProvider';
+import compose from '../../../libs/compose';
+import * as ReportUtils from '../../../libs/ReportUtils';
+import networkPropTypes from '../../../components/networkPropTypes';
+import ONYXKEYS from '../../../ONYXKEYS';
+
+export default function (WrappedComponent) {
+    const propTypes = {
```

Comment:
Move propTypes and default props outside of this function

Reply:
@ArekChr There are 2 functions in HOC. One is outside function representing HOC and another function within representing enhanced wrapper component. `propTypes` and `defaultProps` are already outside the enhanced component but they are within the HOC function.
i.e. 

https://github.com/Expensify/App/blob/98b6f7594b9bf88f135609155b19eb348acbf572/src/pages/home/report/withReportAndPrivateNotesOrNotFound.js#L18-L56

i.e. its outside function defined at line 55-56 but inside function defined at line 18-19.

This is inline within other HOC like `withReportOrNotFound` and `withReportAndReportActionOrNotFound`. Example snippet of `withReportOrNotFound`:
 
https://github.com/Expensify/App/blob/651be71d327cc08582e0c7760898d30ac1ae1a77/src/pages/home/report/withReportAndReportActionOrNotFound.js#L18C8-L80

Reply:
I see we have 2 different approaches in the codebase. However, If you define propTypes and defaultProps outside the component, they'll be redefined every time the component is invoked, which is not efficient.

Reply:
@ArekChr Sorry for the confusion. I want to ask further clarification. Is your proposal is to move the propTypes and defaultProps inside the component or outside the component?
In first comment, you mentioned moving it outside of the function:
> Move propTypes and default props outside of this function

In new comment, you mentioned that defining the propTypes and defaultProps outside the component makes it inefficient as props are redefined every time the component is invoked.
> I see we have 2 different approaches in the codebase. However, If you define propTypes and defaultProps outside the component, they'll be redefined every time the component is invoked, which is not efficient.


My understanding is that we should not define the propTypes and defaultProps inside component because they will be redefined every time the component is invoked. Is this understanding not correct?

Assuming my understanding is correct, I am defining the propTypes and defaultProps outside the React component. In below example, **the component is defined at line 55**. **The function defined in line 18 is not React component, it's the HOC function.** The propTypes and defaultProps are outside React component where the component is defined in line 55.

https://github.com/Expensify/App/blob/98b6f7594b9bf88f135609155b19eb348acbf572/src/pages/home/report/withReportAndPrivateNotesOrNotFound.js#L18-L56

Let me know if this explanation and understanding correct? Otherwise we can discuss it over GitHub issue or slack thread.

Reply:
Sorry for the confusion, I had a typo in the previous message, I mean inside, if we define propTypes inside HOC these props will be recreated each time per instance, but this is just a little thing, props are not heavy, so we can leave as it is.

Reply:
Thanks for the clarifications. I understand your point now. I have moved it outside.